platform/linux-generic/odp_ipsec_sad.c

```diff
@@ -24,8 +24,47 @@
 #define IPSEC_SA_STATE_FREE	0xc0000000
 #define IPSEC_SA_STATE_RESERVED	0x80000000
 
+/*
+ * To avoid checking and updating the packet and byte counters in the
+ * SA for every packet, we increment the global counters once for several
+ * packets. We decrement a preallocated thread-local quota for every
+ * packet. When the quota runs out, we get a new quota by incementing the
+ * global counter.
+ *
+ * This improves performance but the looser synchronization between
+ * threads makes life time warnings and errors somewhat inaccurate.
+ * The warnings and errors may get triggered a bit too early since
+ * some threads may still have unused quota when the first thread
+ * hits the limit.
```

Comment:
I'd like to ask you to split this patch to a separate pull request:
 - threads can report failure early (e.g. in worst case on 96-core box warning can come up around 6000 patches early)
 - threads can disagree on counters state: one thread can start returning 'overrun' errors, while other threads (still having quota) can process packets freely.